qrenderdoc/Windows/MainWindow.cpp

```diff
@@ -2836,3 +2846,26 @@ void MainWindow::showLaunchError(ReplayStatus status)
   }
   GUIInvoke::call(this, [this, title, message]() { RDDialog::warning(this, title, message); });
 }
+
+bool MainWindow::isCapturableAppRunningOnAndroid(QString packageName)
+{
+  if(!m_Ctx.Replay().CurrentRemote() || !m_Ctx.Replay().CurrentRemote()->IsADB())
+    return false;
+
+  rdcstr host = m_Ctx.Replay().CurrentRemote()->hostname;
+  uint32_t ident = RENDERDOC_EnumerateRemoteTargets(host.c_str(), 0);
+
+  if(ident == 0)
+    return false;
+
+  ITargetControl *conn = RENDERDOC_CreateTargetControl(host.c_str(), ident, "testConnection", false);
+  if(conn == NULL)
+    return false;
+
+  if(QString::fromUtf8(conn->GetTarget()).compare(packageName) != 0)
```

Comment:
Does the package name need to be equal here for it to be a problem? Isn't android limited to a single application at a time due to its socket requirements, so even if a different package is running it would still fail to connect to a new package properly?

Reply:
If we launch the same app that is already running, I wouldn't display the warning because later it will be force stopped anyway before launching it again.

Reply:
Ah I see, that makes sense. I interpreted the line of code backwards.

It would be good to change this to be `if(QString::fromUtf8(conn->GetTarget()) != packageName)` which is easier to read, if you're not using case insensitive comparison I don't think there's a reason to use `QString::compare`.